core/src/test/kotlin/ch/ergon/dope/buildTest/QueryBuilderTest.kt

```diff
@@ -787,4 +792,21 @@ class QueryBuilderTest {
 
         assertEquals(expected, actual)
     }
+
+    @Test
+    fun `should support selecting case`() {
+        val someBucket = someBucket()
+        val expected = "SELECT CASE `numberField` WHEN `other` THEN `numberField` END, " +
+            "CASE WHEN `booleanField` THEN `numberField` ELSE `stringField` END AS `alias` FROM `someBucket`"
+
+        val actual = create
+            .select(
+                case(someNumberField()).whenThen(someNumberField("other").resultsIn(someNumberField())),
+                whenThen(someBooleanField().resultsIn(someNumberField())).otherwise(someStringField()).alias("alias"),
```

Comment:
This is irritating that a simple whenThen-function will generate CASE ..., cant we just pass _case()_ with no argument?

Reply:
yes you are correct. I thought about changing that but forgot it 🙈